app/services/editions/db/FrontsQueries.scala

```diff
@@ -11,6 +11,15 @@ trait FrontsQueries {
           SET metadata = ${metadata.toPGobject}
           WHERE id = $id
       """.execute().apply()
+
+    sql"""
+        SELECT metadata FROM fronts WHERE id = $id
+      """.map { rs =>
+      rs.stringOpt("metadata").map { metadataString =>
+        // Throw if we can't parse the metadata to signal to the user that something is broken
+        Json.parse(metadataString).validate[EditionsFrontMetadata].get
```

Comment:
`metadata` is already of type `EditionsFrontMetadata`, so how would validation fail?

Reply:
If the JSON in the database doesn't reflect the case class we expect. For example if a required field is missing after a migration or whatevs.